File: Base Dati/geometria/segmento.py
from punto import Punto
from math import sqrt, pow

'''  '''

class Segmento:
    def __init__(self, A:Punto, B:Punto) -> None:
        self.A = A
        self.B = B

    def lunghezza(self):
        return sqrt(pow(self.B.x - self.A.x, 2) + pow(self.B.y - self.A.y, 2))

    def __str__(self) -> str:
        return f"Segmento di coordinate {self.A} - {self.B}; ha lunghezza {self.lunghezza()}"

File: Java/Prj05_GeometriaPiana/src/figure/Segmento.java
package figure;

public class Segmento extends LuogoGeometrico{

    private Punto a,b;

    public Segmento(Punto a, Punto b) {
        this.nome = "Segmento";
        this.a = a;
        this.b = b;
    }

    public double lunghezza(){
        return Math.sqrt(Math.pow(a.getX() - b.getX(),2) + Math.pow(a.getY() - b.getY(),2));
    }

    @Override
    public String toString() {
        return super.toString() +
                " [a=" + a +
                ", b=" + b +
                ", Lunghezza=" + lunghezza() +
                ']';
    }


}
